pythonforandroid/recipes/Pillow/__init__.py

```diff
@@ -0,0 +1,72 @@
+from pythonforandroid.recipe import CompiledComponentsPythonRecipe
+from pythonforandroid.toolchain import shprint
+from os.path import join, dirname
+import sh
+
+
+class PillowRecipe(CompiledComponentsPythonRecipe):
+
+    version = '5.2.0'
+    url = 'https://github.com/python-pillow/Pillow/archive/{version}.tar.gz'
+    site_packages_name = 'Pillow'
+    depends = [
+        ('python2', 'python3crystax'),
+        'png',
+        'jpeg',
+        'freetype',
+        'setuptools'
+    ]
+    patches = [
+        join('patches', 'fix-docstring.patch'),
+        join('patches', 'fix-setup.patch')
+    ]
+
+    call_hostpython_via_targetpython = False
+
+    def get_recipe_env(self, arch=None):
+        env = super(PillowRecipe, self).get_recipe_env(arch)
+        py_ver = self.ctx.python_recipe.version[0:3]
+
+        ndk_dir = self.ctx.ndk_platform
+        ndk_lib_dir = join(ndk_dir, 'usr', 'lib')
+        ndk_include_dir = (join(self.ctx.ndk_dir, 'sysroot', 'usr', 'include')
+            if py_ver == '2.7' else join(ndk_dir, 'usr', 'include'))
```

Comment:
Was the `zlib.h` include actually deployed here on your Python 2.7 installation? Because on mine it's seems that `join(ndk_dir, 'usr', 'include'))` doesn't exist, while the former does `(join(self.ctx.ndk_dir, 'sysroot', 'usr', 'include')`.
So basically when running recipe on Python2, the `ZLIB_ROOT` would contain the following:
```
/home/andre/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm/usr/lib|/home/andre/.buildozer/android/platform/android-ndk-r9c/sysroot/usr/include
```
The lib directory seems fine:
```
ls /home/andre/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm/usr/lib/
crtbegin_dynamic.o  crtbegin_so.o  crtbegin_static.o  crtend_android.o  crtend_so.o  libandroid.so  libc.a  libc.so  libdl.so  libEGL.so  libGLESv1_CM.so  libGLESv2.so  libGLESv3.so  libjnigraphics.so  liblog.so  libm.a  libm_hard.a  libm.so  libOpenMAXAL.so  libOpenSLES.so  libstdc++.a  libstdc++.so  libthread_db.so  libz.so  rs
```
However the include is not:
```
ls /home/andre/.buildozer/android/platform/android-ndk-r9c/sysroot/usr/include
ls: cannot access '/home/andre/.buildozer/android/platform/android-ndk-r9c/sysroot/usr/include': No such file or directory
```
But if I skip line 33 and set the include as `ndk_include_dir = (join(self.ctx.ndk_dir, 'sysroot', 'usr', 'include')` then it contains the includes:
```
ls /home/andre/.buildozer/android/platform/android-ndk-r9c/platforms/android-19/arch-arm/usr/include/
alloca.h  asm          byteswap.h  dlfcn.h  endian.h  fcntl.h     fnmatch.h  GLES   grp.h       KHR        limits.h  locale.h  math.h    mtd      netinet     OMXAL       poll.h     regex.h   sched.h      sgtty.h   SLES      stdlib.h   sys       termios.h    time.h    utime.h  wctype.h
android   asm-generic  ctype.h     EGL      err.h     features.h  fts.h      GLES2  inttypes.h  lastlog.h  link.h    machine   memory.h  net      netpacket   pathconf.h  pthread.h  resolv.h  semaphore.h  sha1.h    stdint.h  string.h   syslog.h  thread_db.h  unistd.h  utmp.h   zconf.h
arpa      assert.h     dirent.h    elf.h    errno.h   fenv.h      getopt.h   GLES3  jni.h       libgen.h   linux     malloc.h  mntent.h  netdb.h  nsswitch.h  paths.h     pwd.h      rs        setjmp.h     signal.h  stdio.h   strings.h  termio.h  time64.h     util.h    wchar.h  zlib.h
```